packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -32,14 +32,15 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
   const cookies = readRawCookies(opts) || {}
 
   let delay: number | undefined
-  if (opts.maxAge) {
+
+  if (opts.maxAge !== undefined) {
```

Comment:
I think you're accounting for 0 here but we don't do anything with it if so, and it doesn't handle null this way.

```suggestion
  if (opts.maxAge) {
```

Reply:
![image](https://github.com/nuxt/nuxt/assets/48835293/6c2d18a9-d1c3-424b-b7be-63e446ab7ba2)

Type def disallows `null` though, do we need to handle it?

Also, the initial implementation accounted for 0 and set the value to `undefined`. Reverted to it with some minor tweaks. I kind of prefer letting the `cookieRef` handle the delay _regardless of its value_. Open to your thoughts!

Reply:
The below sentence is a sole reason why I thought about making this PR:
![image](https://github.com/nuxt/nuxt/assets/48835293/8338b16f-9a5e-4cef-89e9-24e01e3fa87d)

Reply:
No, that should definitely be the case, but I'm not sure that `cookieRef` will do that as-is (or is needed to do that).

Reply:
Shall we instead add a handler for that so we do it directly after we initialise the ref?

Reply:
Just pushed an update which I hope is closer to the desirable solution. Let me know your thoughts